File: statsig-pyo3/py_src/statsig_python_core/observability_client.py
from statsig_python_core import ObservabilityClientBase
from typing import Optional, Dict, Tuple


class ObservabilityClient(ObservabilityClientBase):
    def __init__(self):
        super().__init__()
        self.init_fn = self.init
        self.increment_fn = self.increment
        self.gauge_fn = self.gauge
        self.dist_fn = self.dist
        self.error_fn = self.error
        self.should_enable_high_cardinality_for_this_tag_fn = self.should_enable_high_cardinality_for_this_tag

    def init(self):
        pass

    def increment(self, metric_name: str, value: float, tags: Optional[Dict[str, str]] = None):
        pass

    def gauge(self, metric_name: str, value: float, tags: Optional[Dict[str, str]] = None):
        pass

    def dist(self, metric_name: str, value: float, tags: Optional[Dict[str, str]] = None):
        pass

    def error(self, tag: str, error: str):
        pass

    def should_enable_high_cardinality_for_this_tag(self, tag: str):
        pass
    

File: statsig-ffi/bindings/java/src/main/java/com/statsig/ObservabilityClient.java
package com.statsig;

import java.util.Map;

public interface ObservabilityClient {
    void init();
    void increment(String metricName, double value, Map<String, String> tags);
    void gauge(String metricName, double value, Map<String, String> tags);
    void dist(String metricName, double value, Map<String, String> tags);
    void error(String tag, String errorMessage);
}

